vite.config.js

```diff
@@ -53,4 +53,5 @@ export default defineConfig(function ({ mode }) {
       },
     },
   }
-})
+}
+export default defineConfig(({ mode }) => config({ mode }))
```

Comment:
**Issue**: This prevents all type checking that is provided through `defineConfig` to happen when defining `config`.

I see why you’re doing it like that, but please somehow restore the type checking for the `config` object. Ideally, the development config should have the same type checking capabilities. In the development config, the parameters of the config function (e.g. `mode`) are now completely untyped, too.

Reply:
I moved the `vite.config` files to typescript, no idea why they weren't typescript already, guessing having these as typescript files is fine and all seems fine to me. Little note here: later on we could completely remove `mode` and just use the separate files, but we can sort that once we have all this finished up.